app/controllers/candidate_interface/prefill_application_form_controller.rb

```diff
@@ -54,5 +49,36 @@ def destroy_blank_application
       application_form = current_candidate.application_forms.first
       application_form.destroy if application_form.blank_application?
     end
+
+    def test_application_options
+      test_application_options = {
+        recruitment_cycle_year: RecruitmentCycle.current_year,
+        states: [:unsubmitted_with_completed_references],
+        courses_to_apply_to: Course.current_cycle.open_on_apply.joins(:course_options).merge(CourseOption.available),
+        candidate: current_candidate,
+      }
+
+      data = read_prefill_application_data
+
+      if data
+        course_from_find = Course.find(data[:course_id])
+        test_application_options.merge!(courses_to_apply_to: [course_from_find])
+        clear_prefill_application_data
+      end
+
+      test_application_options
+    end
+
+    def read_prefill_application_data
+      Rails.cache.read(key)
+    end
+
+    def clear_prefill_application_data
+      Rails.cache.clear(key)
+    end
+
+    def key
+      "prefill_application_#{current_candidate.id}"
```

Comment:
We could probably refactor this key generation and the cache read/write so they could be reused across this controller and the after_sign_in controller? Not essential, but would insure against bugs where we changed the key in one place and not the other.

Reply:
Yep good point - I've extracted the caching logic to a class - https://github.com/DFE-Digital/apply-for-teacher-training/pull/5375/commits/40a614f4c70b7d675ea30600e2624f50d6d2f3db